### PR TITLE
8059 add new "zfstest" test cases to stress refquota

### PR DESCRIPTION
**Pull request by prakashsurya**
Reviewed by: Sonu Pillai <sonu.pillai@delphix.com>
Reviewed by: John Kennedy <john.kennedy@delphix.com>
Reviewed by: Christian Hellman <christian.hellman@delphix.com>
Reviewed by: Dan Kimmel <dan.kimmel@delphix.com>
Reviewed by: Sebastien Roy <sebastien.roy@delphix.com>

Upstream bugs: 23164, DLPX-48406

### CODE REVIEW

**Comment by prakashsurya** [comment]
@zettabot go

File: usr/src/test/zfs-tests/tests/functional/refquota/refquota_007_neg.ksh [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/bin/ksh
+#
+# CDDL HEADER START
```
Comment (yuripv) [inline usr/src/test/zfs-tests/tests/functional/refquota/refquota_007_neg.ksh]: Can we use CDDL text from usr/src/prototypes for new files please?

File: usr/src/test/zfs-tests/tests/functional/refquota/refquota_007_neg.ksh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+verify_runnable "both"
+
+function cleanup
+{
+	log_must zfs destroy -rf $TESTPOOL/$TESTFS
+	log_must zfs create $TESTPOOL/$TESTFS
+	log_must zfs set mountpoint=$TESTDIR $TESTPOOL/$TESTFS
+}
+
+log_assert "refquota limits the amount of space a dataset can consume, " \
```
Comment (yuripv) [inline usr/src/test/zfs-tests/tests/functional/refquota/refquota_007_neg.ksh]: Don't need whitespace inside the string, and ```\```.

File: usr/src/test/zfs-tests/include/default.cfg [hunk 3]
```diff
@@ -88,6 +88,11 @@ export TESTSUBFS=subfs
 export TESTSUBFS1=subfs1
 export TESTSUBFS2=subfs2
 
+# some test sub file system names
+export TESTSUBFS=subfs.$$
+export TESTSUBFS1=subfs1.$$
+export TESTSUBFS2=subfs2.$$
+
```
Comment (jwk404) [inline usr/src/test/zfs-tests/include/default.cfg]: The trailing ".$$" on these files isn't necessary, and make it more difficult to do cleanup of ad hoc testing. They've been removed from the Delphix version.

Comment (jwk404) [inline usr/src/test/zfs-tests/include/default.cfg]: Actually, looking at the diffs, the lines above this addition are the correct ones.

**Comment by prakashsurya** [comment]
@zettabot go

**Pull request closed without merging**